online-shop/src/app/product-details/product-details.component.ts

```diff
@@ -1,22 +1,34 @@
-import {Component, OnInit} from '@angular/core';
-import {ActivatedRoute} from "@angular/router";
-import {Product} from "../../product";
-import {products} from "../../product";
-
+import { Component, OnInit } from '@angular/core';
+import {ActivatedRoute, Router} from '@angular/router';
+import { Product } from '../../product';
+import {ProductService} from "../product.service";
 
 @Component({
   selector: 'app-product-details',
   templateUrl: './product-details.component.html',
-  styleUrls: ['./product-details.component.scss']
+  styleUrls: ['./product-details.component.scss'],
 })
 export class ProductDetailsComponent implements OnInit {
   id = 0;
-  product!: Product
-constructor(
-  private route: ActivatedRoute,
-) {}
+  product!: Product;
+  constructor(private route: ActivatedRoute, private productService: ProductService) {}
+
+
+
   ngOnInit(): void {
-   this.id = parseInt(this.route.snapshot.paramMap.get('id')!);
-   this.product = products.find(product => product.id === this.id)!;
+    this.id = parseInt(this.route.snapshot.paramMap.get('id')!);
+    this.productService.getProductById(this.id).subscribe(data => {
+      this.product = data;
+    })
   }
+
+deleteProduct() {
```

Comment:
since it is typeScript, and types are the gimmick, all variables and functions should have a type specified.
A thing that will help you spot all the functions and vars that doesn't have types. And in the future will inform you when you missed one, since you added eslint and prettier, let's use them -> https://typescript-eslint.io/rules/explicit-function-return-type/